sim/mage/talents.go

```diff
@@ -1,74 +1,140 @@
 package mage
 
 import (
+	"slices"
 	"time"
 
 	"github.com/wowsims/sod/sim/core"
 	"github.com/wowsims/sod/sim/core/stats"
 )
 
 func (mage *Mage) ApplyTalents() {
-	mage.applyIgnite()
-	mage.applyArcaneConcentration()
-	mage.applyMasterOfElements()
-	mage.applyWintersChill()
+	mage.applyArcaneTalents()
+	mage.applyFireTalents()
+	mage.applyFrostTalents()
+}
 
-	mage.registerArcanePowerCD()
+func (mage *Mage) applyArcaneTalents() {
+	mage.applyArcaneConcentration()
 	mage.registerPresenceOfMindCD()
-	mage.registerCombustionCD()
-	mage.registerColdSnapCD()
-
-	mage.PseudoStats.SpiritRegenRateCasting += 0.05 * float64(mage.Talents.ArcaneMeditation)
+	mage.registerArcanePowerCD()
 
-	if mage.Talents.ElementalPrecision > 0 {
-		bonusHit := 2 * float64(mage.Talents.ElementalPrecision) * core.SpellHitRatingPerHitChance
+	// Arcane Subtlety
+	if mage.Talents.ArcaneSubtlety > 0 {
+		threatMultiplier := .20 * float64(mage.Talents.ArcaneSubtlety)
 		mage.OnSpellRegistered(func(spell *core.Spell) {
-			if spell.SpellSchool.Matches(core.SpellSchoolFire) || spell.SpellSchool.Matches(core.SpellSchoolFrost) {
-				spell.BonusHitRating += bonusHit
+			if spell.SpellSchool.Matches(core.SpellSchoolArcane) && spell.Flags.Matches(SpellFlagMage) {
+				spell.ThreatMultiplier *= threatMultiplier
 			}
 		})
 	}
 
+	// Arcane Focus
 	if mage.Talents.ArcaneFocus > 0 {
 		bonusHit := 2 * float64(mage.Talents.ArcaneFocus) * core.SpellHitRatingPerHitChance
 		mage.OnSpellRegistered(func(spell *core.Spell) {
-			if spell.SpellSchool.Matches(core.SpellSchoolArcane) {
+			if spell.SpellSchool.Matches(core.SpellSchoolArcane) && spell.Flags.Matches(SpellFlagMage) {
 				spell.BonusHitRating += bonusHit
 			}
 		})
 	}
 
-	if mage.Talents.PiercingIce > 0 {
-		mage.PseudoStats.SchoolDamageDealtMultiplier[stats.SchoolIndexFrost] *= 1 + 0.02*float64(mage.Talents.PiercingIce)
+	// Magic Absorption
+	if mage.Talents.MagicAbsorption > 0 {
+		magicAbsorptionBonus := 2 * float64(mage.Talents.MagicAbsorption)
+		mage.AddStat(stats.ArcaneResistance, magicAbsorptionBonus)
+		mage.AddStat(stats.FireResistance, magicAbsorptionBonus)
+		mage.AddStat(stats.FrostResistance, magicAbsorptionBonus)
+		mage.AddStat(stats.NatureResistance, magicAbsorptionBonus)
+		mage.AddStat(stats.ShadowResistance, magicAbsorptionBonus)
 	}
 
-	if mage.Talents.FirePower > 0 {
-		mage.PseudoStats.SchoolDamageDealtMultiplier[stats.SchoolIndexFire] *= 1 + 0.02*float64(mage.Talents.FirePower)
+	// Arcane Meditation
+	mage.PseudoStats.SpiritRegenRateCasting += 0.05 * float64(mage.Talents.ArcaneMeditation)
+
+	if mage.Talents.ArcaneMind > 0 {
+		mage.MultiplyStat(stats.Mana, 1.0+0.02*float64(mage.Talents.ArcaneMind))
 	}
 
+	// Arcane Instability
+	mage.AddStat(stats.SpellCrit, float64(mage.Talents.ArcaneInstability)*1*core.SpellCritRatingPerCritChance)
+	mage.PseudoStats.DamageDealtMultiplier *= 1 + .01*float64(mage.Talents.ArcaneInstability)
```

Comment:
This is a spell mod, it only affects mage spells and is additive with other spell mods.
See https://www.wowhead.com/classic/spell=15060/arcane-instability

Reply:
Updated! Let me know if I did something wrong 😄